src/urlHelpers.js

```diff
@@ -32,7 +32,7 @@ module.exports.getBaseUrl = function(req, protocol) {
   return url.format({
     protocol: urlProtocol || 'https',
     host: req.headers.host,
-    pathname: originalUrl.replace(req.path, '').replace(/\/$/g, '')
+    pathname: originalUrl.replace(new RegExp(req.path + '$'), '').replace(/\/$/g, '')
```

Comment:
The intention here is to remove the path only at the end of the  url.

Reply:
Does req.path is sanitized for being use in a regex ? Each character in req.path would be interpreted as part of the regex.

```
let r = new RegExp('/some-urls/asdfsdf.html/?quwer=sadf&test[]=123$')
r.test('/some-urls/asdfsdf.html/?quwer=sadf&test[]=123') --> false
```